server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/codeSearch.test.ts

```diff
@@ -0,0 +1,202 @@
+import * as assert from 'assert'
+import { CodeSearch, CodeSearchOutput } from './codeSearch'
+import { testFolder } from '@aws/lsp-core'
+import * as path from 'path'
+import * as fs from 'fs/promises'
+import { TestFeatures } from '@aws/language-server-runtimes/testing'
+import { Features } from '@aws/language-server-runtimes/server-interface/server'
+import { LocalProjectContextController } from '../../../shared/localProjectContextController'
+import { Chunk } from 'local-indexing'
+import { stub, restore, SinonStub } from 'sinon'
+
+describe('CodeSearch Tool', () => {
+    let tempFolder: testFolder.TestFolder
+    let testFeatures: TestFeatures
+    let mockLocalProjectContextController: Partial<LocalProjectContextController>
+    let getInstanceStub: SinonStub
+
+    before(async () => {
+        testFeatures = new TestFeatures()
+        // @ts-ignore does not require all fs operations to be implemented
+        testFeatures.workspace.fs = {
```

Comment:
nit: I think you can do `testFeatures.workspace.fs.exists = ` then you don't need the `// @ts-ignore`?

Reply:
updated